lib/main/server.js

```diff
@@ -0,0 +1,50 @@
+var http = require('http');
+var url = require('url');
+var Scrapper = require('./scrapper.js');
+
+function Server() {
+
+  this.start = function() {
+    http.createServer(function(request, response) {
+      var urlParams = url.parse(request.url, true);
+      if (request.method === 'GET' && urlParams.pathname === '/I/want/title/') {
+        var addresses = (typeof(urlParams.query.address) == 'string') ? [urlParams.query.address] : urlParams.query.address;
+        var titles = [];
+        var body = [];
+
+        function scrapTitle(index, callback) {
+          new Scrapper(addresses[index], function(result){
+            callback(result.url + " - " + result.dom('title').text(), index);
+          }, function(error){
+            callback(error.url + " - NO RESPONSE", index);
+          });
+        }
+
+        function processTitle(title, index){
+          titles.push(title);
+          if(index < addresses.length - 1){
+              scrapTitle(index + 1, processTitle);
+          } else {
+            response.statusCode = 200;
```

Comment:
Above if and code in this if are at completely different level. First if is business logic and second is rendering response. A separate function for rendering view can help write code at same level and will improve readability.